src/codec/PairingCode.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Schema } from "@project-chip/matter.js";
+import { Base38 } from "./Base38";
+import { BitField, BitFieldEnum, BitFlag, ByteArrayBitmapSchema, TypeFromBitmapSchema, BitmapSchema } from "./BitmapSchema";
+import { Verhoeff } from "../math/Verhoeff";
+
+/** See {@link MatterCoreSpecificationV1_0} § 5.1.3.1 Table 35 */
+export enum CommissionningFlowType {
+    /** When uncom­missioned, the device always enters commissioning mode upon power-up. */
+    Standard = 0,
+
+    /** User action required to enter commissioning mode. */
+    UserIntent = 1,
+
+    /** Interaction with a vendor-specified means is needed before commissioning. */
+    Custom = 2,
+}
+
+/** See {@link MatterCoreSpecificationV1_0} § 5.1.3.1 Table 36 */
+export const DiscoveryCapabilitiesSchema = BitmapSchema({
+    /** Device supports hosting a Soft-AP when not commissioned. */
+    softAccessPoint: BitFlag(0),
+
+    /** Device supports BLE for discovery when not commissioned. */
+    ble: BitFlag(1),
+
+    /** Device is already on the IP network. */
+    onIpNetwork: BitFlag(2),
+});
+
+/** See {@link MatterCoreSpecificationV1_0} § 5.1.3.1 Table 35 */
+const QrCodeDataSchema = ByteArrayBitmapSchema({
```

Comment:
Which mechanism adds the 4 bytes at the end of the struct? meaning the "Padding" from specs

Reply:
The ByteArrayBitmapSchema sets to 0 all bits not defined.
Yes, I know: probably we should add a mechanism to explicitly declared reserve bits and their values...